lib/swarm/tracker/tracker.ex

```diff
@@ -698,11 +725,9 @@ defmodule Swarm.Tracker do
                   debug "#{inspect name} has requested to be restarted"
                   {:ok, new_state} = remove_registration(obj, %{state | clock: lclock})
                   send(pid, {:swarm, :die})
-                  case handle_call({:track, name, m, f, a}, nil, %{state | clock: lclock}) do
```

Comment:
We should use the new state returned from the last call which manipulates the state (i.e. the clock should be incremented twice)